src/Esfa.Vacancy.Register.Api/Mappings/VacancyMappings.cs

```diff
@@ -1,33 +1,124 @@
-﻿using AutoMapper;
+﻿using System.Globalization;
+using AutoMapper;
+using Esfa.Vacancy.Api.Types;
+using Esfa.Vacancy.Register.Domain.Entities;
 
 namespace Esfa.Vacancy.Register.Api.Mappings
 {
     public class VacancyMappings : Profile
     {
+        private const string UnknownText = "Unknown";
+
         public VacancyMappings()
         {
             CreateMap<Domain.Entities.Vacancy, Vacancy.Api.Types.Vacancy>()
                 .ForMember(apiType => apiType.VacancyUrl, opt => opt.Ignore())
+                .ForMember(apiType => apiType.Wage, opt => opt.Ignore())
                 .ForMember(apiType => apiType.VacancyType, opt => opt.MapFrom(source => source.VacancyTypeId))
                 .ForMember(apiType => apiType.WageUnit, opt => opt.MapFrom(source => source.WageUnitId))
                 .ForMember(apiType => apiType.VacancyReference, opt => opt.MapFrom(source => source.VacancyReferenceNumber))
                 .ForMember(apiType => apiType.LocationType, opt => opt.MapFrom(source => source.VacancyLocationTypeId))
                 .AfterMap((src, dest) =>
                 {
-                    if (src.IsAnonymousEmployer && (Domain.Entities.VacancyStatus)src.VacancyStatusId == Domain.Entities.VacancyStatus.Live)
+                    if (src.IsAnonymousEmployer && (VacancyStatus)src.VacancyStatusId == VacancyStatus.Live)
                     {
                         ApplyAnonymisationToVacancy(src, dest);
                     }
+
+                    if (dest.VacancyType == VacancyType.Traineeship)
+                    {
+                        ResetWageFieldsForTraineeship(dest);
+                    }
+                    else
+                    {
+                        MapWageFields(src, dest);
+                    }
                 });
         }
-        private static void ApplyAnonymisationToVacancy(Domain.Entities.Vacancy src, Vacancy.Api.Types.Vacancy dest)
+
+        private void MapWageFields(Domain.Entities.Vacancy src, Vacancy.Api.Types.Vacancy dest)
+        {
+            switch (src.WageType)
+            {
+                case (int) WageType.LegacyText:
+                    dest.Wage = UnknownText;
+                    break;
+                case (int) WageType.LegacyWeekly:
+                case (int)WageType.Custom:
+                    dest.Wage = GetFormattedCurrencyString(src.WeeklyWage) ?? UnknownText;
+                    break;
+                case (int) WageType.ApprenticeshipMinimum:
+                    dest.Wage = src.MinimumWageRate.HasValue && src.HoursPerWeek.HasValue 
+                                ? GetFormattedCurrencyString(src.MinimumWageRate.Value * src.HoursPerWeek.Value)
+                                : UnknownText;
```

Comment:
Maybe move this to a function so it looks similar to the other cases.

Reply:
Good call.